test/expect/pure/formats/query/query.ml

```diff
@@ -21,6 +21,14 @@ let%expect_test _ =
   query "a" "a=b&a=c";
   query "c" "a=b&c=d";
   query "a b" "a+b=c";
+  query "a" "a";
+  query ~filter_empty:true "a" "a";
+  query ~filter_empty:true "a" "a ";
+  query ~filter_empty:true "a" "a  ";
+  query ~filter_empty:true "a" "a=";
+  query ~filter_empty:true "a" "a= ";
+  query ~filter_empty:true "a" "a=  ";
+  query ~filter_empty:true "a" "a= &a=b";
```

Comment:
Can you add `query ~filter_empty:true "a" "a= b "` and any other tests that use `~filter_empty` but should *not* produce `None`?